samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemAdmin.java

```diff
@@ -0,0 +1,154 @@
+  private final InMemorySystemConfig inMemorySystemConfig;
+
+  public InMemorySystemAdmin(InMemoryManager manager, Config config) {
+    inMemoryManager = manager;
+    inMemorySystemConfig = new InMemorySystemConfig(config);
+  }
+
+  @Override
+  public void start() {
+
+  }
+
+  @Override
+  public void stop() {
+
+  }
+
+  /**
+   * Fetches the offsets for the messages immediately after the supplied offsets
+   * for a group of SystemStreamPartitions.
+   *
+   * @param offsets
+   *          Map from SystemStreamPartition to current offsets.
+   * @return Map from SystemStreamPartition to offsets immediately after the
+   *         current offsets.
+   */
+  @Override
+  public Map<SystemStreamPartition, String> getOffsetsAfter(Map<SystemStreamPartition, String> offsets) {
+    return offsets.entrySet()
+        .stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, entry -> {
+            String offset = entry.getValue();
+            return String.valueOf(Integer.valueOf(offset) + 1);
+          }));
+  }
+
+  /**
+   * Fetch metadata from a system for a set of streams.
+   *
+   * @param streamNames
+   *          The streams to to fetch metadata for.
+   * @return A map from stream name to SystemStreamMetadata for each stream
+   *         requested in the parameter set.
+   */
+  @Override
+  public Map<String, SystemStreamMetadata> getSystemStreamMetadata(Set<String> streamNames) {
+    return inMemoryManager.getSystemStreamMetadata(streamNames);
+  }
+
+  /**
+   * Compare the two offsets. -1, 0, +1 means offset1 &lt; offset2,
+   * offset1 == offset2 and offset1 &gt; offset2 respectively. Return
+   * null if those two offsets are not comparable
+   *
+   * @param offset1 First offset for comparison.
+   * @param offset2 Second offset for comparison.
+   * @return -1 if offset1 &lt; offset2; 0 if offset1 == offset2; 1 if offset1 &gt; offset2. Null if not comparable
+   */
+  @Override
+  public Integer offsetComparator(String offset1, String offset2) {
+    int o1 = Optional.ofNullable(offset1)
+        .filter(Strings::isNullOrEmpty)
+        .map(Integer::parseInt)
+        .orElse(-1);
+
+    int o2 = Optional.ofNullable(offset2)
+        .filter(Strings::isNullOrEmpty)
+        .map(Integer::parseInt)
+        .orElse(-1);
+
+    if (o1 == -1 || o2 == -1) {
+      return null;
+    }
+
+    return Integer.compare(o1, o2);
+  }
+
+  /**
+   * Create a stream described by the spec.
+   *
+   * @param streamSpec  The spec, or blueprint from which the physical stream will be created on the system.
+   * @return            {@code true} if the stream was actually created and not pre-existing.
+   *                    {@code false} if the stream was pre-existing.
+   *                    A RuntimeException will be thrown if creation fails.
+   */
+  @Override
+  public boolean createStream(StreamSpec streamSpec) {
+    return Optional.ofNullable(inMemorySystemConfig.getSerializedDataSet(streamSpec.getId()))
```

Comment:
I think you should just do a inMemoryManager.intializeStream(streamSpec, serializedData). Do a null check inside the function.